visualization/app/codeCharta/ui/artificialIntelligence/artificialIntelligence.component.ts

```diff
@@ -103,6 +103,12 @@ export class ArtificialIntelligenceController implements FilesSelectionSubscribe
 	}
 
 	private calculate() {
+		const { experimentalFeaturesEnabled } = this.storeService.getState().appSettings
```

Comment:
This won't work: If the web application is loaded with a clear local storage, then the experimental features are disabled by default. Turning them on in the global settings will not change the value of the constant `experimentalFeaturesEnabled `. Instead, please subscribe the state by `ExperimentalFeaturesEnabledSubscriber`.

Reply:
I am sorry, I was wrong! You're totally right that you always have the current enabled/disabled value.
My (manual) test failed, because the calculation has not been triggered on activating the experimental features.

That means, keep the if as it is and we just have to subscribe the `ExperimentalFeaturesEnabledSubscriber` and call the `debounceCalculation()` method.